huggingface_bot.py

```diff
@@ -0,0 +1,64 @@
+"""
+
+Bot that lets you talk to conversational models available on HuggingFace.
+
+"""
+from __future__ import annotations
+
+from typing import AsyncIterable
+
+from fastapi_poe import PoeBot
+from fastapi_poe.types import QueryRequest
+from huggingface_hub import InferenceClient
+from sse_starlette.sse import ServerSentEvent
+
+
+class HuggingFaceBot(PoeBot):
+    """This bot uses the HuggingFace Inference API.
+
+    By default, it uses the HuggingFace public Inference API, but you can also
+    use this class with a self hosted Inference Endpoint.
+    For more information on how to create a self hosted endpoint, see:
+    https://huggingface.co/blog/inference-endpoints
+
+    Arguments:
+        - model: either the name of the model (if you want to use the public API)
+        or a link to your hosted inference endpoint.
+
+    """
+
+    def __init__(self, model: str, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        if model is None:
+            raise ValueError("Unsupported model: {model}")
+        self.client = InferenceClient(model=model)
+
+    async def query_hf_model(
+        self, current_message_text, bot_messages: list[str], user_messages: list[str]
+    ) -> tuple[int, dict]:
+        return self.client.conversational(
```

Comment:
Since this isn't async, this will block your server and you can't handle multiple simultaneous requests. Maybe that's OK for an example though.